src/Workspaces/CSharp/Portable/Simplification/Simplifiers/ExpressionSimplifier.cs

```diff
@@ -38,6 +38,23 @@ private ExpressionSimplifier()
             out TextSpan issueSpan,
             CancellationToken cancellationToken)
         {
+            replacementNode = null;
+            issueSpan = default;
+
+            if (expression is MemberAccessExpressionSyntax memberAccessExpression &&
+                memberAccessExpression.Expression.IsKind(SyntaxKind.ThisExpression))
+            {
+                if (!MemberAccessExpressionSimplifier.Instance.ShouldSimplifyThisMemberAccessExpression(
+                        memberAccessExpression, semanticModel, options, out var thisExpression, out _, cancellationToken))
+                {
+                    return false;
+                }
+
+                replacementNode = memberAccessExpression.GetNameWithTriviaMoved();
+                issueSpan = thisExpression.Span;
+                return true;
+            }
```

Comment:
dedicated check for this case, instead of being interspersed with many other cases.